### PR TITLE
Add table size on record insert

### PR DESCRIPTION
**Pull request by thorfour**
Change how record sizes are calculated by using an approximation based on number of rows in the slice and the overall record size

### CODE REVIEW

**Comment by asubiotto** [comment]
Would it make sense to add regression test that checks that the expected sizes grow by a reasonable amount when inserting data?

File: bufutils/arrow.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package bufutils
+
+import (
+	"github.com/apache/arrow/go/v10/arrow"
+)
+
+// RecordSize returns the arrow record size in bytes.
+func RecordSize(r arrow.Record) int64 {
```
Comment (asubiotto) [inline bufutils/arrow.go:8]: nit: this might make more sense in the `arrowutils` package

File: table_test.go [hunk 2]
```diff
@@ -1419,3 +1420,97 @@ func Test_RowWriter(t *testing.T) {
...
+
+// Test_Table_Size ensures the size of the table increases by the size of the inserted data.
+func Test_Table_Size(t *testing.T) {
+	test := func(isArrow bool) {
+		c, table := basicTable(t)
+		defer c.Close()
+
+		before := table.ActiveBlock().Size()
+
+		samples := dynparquet.Samples{{
```
Comment (asubiotto) [inline table_test.go:1432]: nit: why not just use `dynparquet.NewTestSamples` and modify those if you need to? It's less verbose

**Review by asubiotto — APPROVED** [review]
LGTM minus nits

File: table_test.go (hunk 2 above)
Comment (thorfour) [reply table_test.go:1432]: simply forgot about it

File: bufutils/arrow.go (hunk 1 above)
Comment (thorfour) [reply bufutils/arrow.go:8]: bah humbug I forgot about that

**Pull request merged**